src/components/composed/StateManager.svelte

```diff
@@ -29,6 +29,11 @@
 
   async function initialize(netId) {
     if (!initStarted && $addressStore !== undefined && netId !== lastConnection.chainId) {
+      // Reload the page if the vaults were previously loaded and the user is switching chains
+      if (!$vaultsLoading && netId !== lastConnection.chainId) {
+        location.reload();
```

Comment:
This is just doing a blatant refresh of the entire page which leads to a new batch or API requests being sent to the respective endpoints. We need a more elegant solution that doesn't create additional queries to keep within API request limits.

Reply:
I've removed this solution and fixed it by more thoroughly removing the errors that were causing this issue. Please see the latest commit for more details.